src/main/java/de/tum/in/www1/artemis/repository/iris/IrisChatSessionRepository.java

```diff
@@ -14,15 +16,41 @@
  */
 public interface IrisChatSessionRepository extends JpaRepository<IrisChatSession, Long> {
 
-    Optional<IrisChatSession> findByExerciseIdAndUserId(Long exerciseId, Long userId);
+    @Query("""
+                SELECT s
+                FROM IrisChatSession s
+                WHERE s.exercise.id = :exerciseId
+                AND s.user.id = :userId
+            """)
+    Optional<List<IrisChatSession>> findByExerciseIdAndUserId(Long exerciseId, Long userId);
 
     @NotNull
-    default IrisChatSession findByExerciseIdAndUserIdElseThrow(long exerciseId, long userId) throws EntityNotFoundException {
+    default List<IrisChatSession> findByExerciseIdAndUserIdElseThrow(long exerciseId, long userId) throws EntityNotFoundException {
         return findByExerciseIdAndUserId(exerciseId, userId).orElseThrow(() -> new EntityNotFoundException("Iris Session"));
     }
 
     @NotNull
     default IrisChatSession findByIdElseThrow(long sessionId) throws EntityNotFoundException {
         return findById(sessionId).orElseThrow(() -> new EntityNotFoundException("Iris Session", sessionId));
     }
+
+    default IrisChatSession findNewestByExerciseIdAndUserIdElseThrow(long exerciseId, long userId) throws EntityNotFoundException {
+        var result = findNewestByExerciseIdAndUserIdHelper(exerciseId, userId);
+        if (result.isEmpty()) {
+            throw new EntityNotFoundException("Iris Session");
+        }
+        if (result.get().size() == 0) {
+            throw new EntityNotFoundException("Iris Session");
+        }
+        return result.get().get(0);
+    }
+
+    @Query("""
+                SELECT s
+                FROM IrisChatSession s
+                WHERE s.exercise.id = :exerciseId
+                AND s.user.id = :userId
+                ORDER BY s.creationDate DESC
+            """)
+    Optional<List<IrisChatSession>> findNewestByExerciseIdAndUserIdHelper(Long exerciseId, Long userId);
```

Comment:
Having a List in an Optional does not make sense here. If no results are found, the list would just be empty